engine/src/main/scala/cromwell/engine/io/nio/NioFlow.scala

```diff
@@ -152,21 +191,21 @@ class NioFlow(parallelism: Int,
 
   private def createDirectories(path: Path) = path.parent.createDirectories()
 
-  private def getFileHashForDrsPath(drsPath: DrsPath): IO[String] = {
```

Comment:
I moved this function into `DrsPath`. There, I removed the wrapping `IO` so as to 1. leave `DrsPath` un-opinionated about cats-effect and 2. to reduce the risk of accidentally stubbing this in tests with a non-delayed `IO` (which affects behavior on retry; I ran into this myself while writing the tests). Other than that and changing `drsPath` references to `this`, the function is unchanged... except for a comment I removed; see below.